pages/cartPage.js

```diff
@@ -0,0 +1,296 @@
+import Head from 'next/head';
+import React from 'react';
+import Header from '../components/Header.js';
+import Footer from '../components/Footer.js';
+import Link from 'next/link';
+import cookie from 'js-cookie';
+import nextCookies from 'next-cookies';
+import RemoveFromCart from '../components/RemoveFromCart.tsx';
+import AddOneItem from '../components/AddOneItem.tsx';
+import ReduceOneItem from '../components/ReduceOneItem.tsx';
+
+export function totalSum(itemsInCart) {
+  return itemsInCart.length !== 0
+    ? itemsInCart.reduce((acc, cur) => {
+        return acc + cur.price;
+      }, 0)
+    : 0;
+}
+function CartPage({ cart, products }) {
+  const itemsInCart = cart || []; //changed it to state const, but it wasnt one before. mb i wanna remove set cookies from other child functions????
+  // const [total, setTotal] = useState(cookie.getJSON('total') || 0);
+
+  //to sync the prop to state when re-rendering happens,
+  //you need a useEffect inside the child that listens to prop change:
+
+  // useEffect(() => {
+  //   cookie.set('cart', itemsInCart);
+  // }, [itemsInCart]); ////this gets run every time the value is changed, so it stores new value
+
+  // const [pieces, setPieces] = useState(0);
+  // const [itemTotal, setItemTotal] = useState(0); //yes haha
+  //here you cant go like itemmm
+  const totalCart = totalSum(itemsInCart);
+  cookie.set('total', totalCart);
+  //       const totalCart =
+  //   itemsInCart.length !== 0
+  //     ? itemsInCart.reduce((acc, cur) => {
+  //         return acc + cur.price;
+  //       }, 0)
+  //     : 0;
+  // cookie.set('total', totalCart);
+  // setTotal(totalCart);
+  // cookie.set('total', total);
+
+  // useEffect(() => {
+  //   cookie.set('total', total);
+  // }, [total]);
+
+  return (
+    <div>
+      <Head>
+        <title>Bist du Teppich: Your Cart </title>
+        <link rel="icon" href="/favicon.png" />
+      </Head>
+      <Header />
+      <div className="container">
+        <h1>Shopping Cart</h1>
+        <Link href="/products">
+          <a className="backToShop"> Back to shop</a>
+        </Link>
+        <div className="tableItems">
+          <div className="headings">
+            <h4> Product</h4>
+            <h4>Description</h4>
+            <h4>Quantity</h4>
+            <h4>Price</h4>
+            <h4>&nbsp;</h4>
+          </div>
+          {itemsInCart.length !== 0
+            ? itemsInCart.map((item) => {
+                return (
+                  <div data-cy={'item-cart'} className="item" key={item.id}>
+                    <img src={item.img} alt="item" />
+                    <p>{item.name}</p>
+                    <span data-cy={'amount-cart'} className="buttonz">
```

Comment:
The style `prop={'value'}` can also be written without the curlies and just with double quotes: `prop="value"`.